### PR TITLE
Add `ImageChoiceWidget`

### PR DESCRIPTION
**Pull request by Pikkupomo**
Also some minor plumbing to allow testing this feature without ui part.

Refs SHOOP-1278

### CODE REVIEW

**Comment by akx** [comment]
Can you rebase? Also, this seems to add a widget _and_ some extra functionality in the media browser view?

**Comment by Pikkupomo** [comment]
rebased and added comment about the extra functionality to commit message

File: shoop/admin/modules/media/views.py [hunk 1]
```diff
@@ -82,6 +84,8 @@ def get_context_data(self, **kwargs):
         return context
 
     def get(self, request, *args, **kwargs):
+        self.filter = request.REQUEST.get("filter")
```
Comment (akx) [inline shoop/admin/modules/media/views.py]: You should change line 82 to also use `self.filter`, now that we have that.

File: shoop/admin/modules/products/views/forms.py [hunk 2]
```diff
@@ -268,6 +268,8 @@ class ProductImageMediaForm(BaseProductMediaForm):
 
     def __init__(self, **kwargs):
         super(ProductImageMediaForm, self).__init__(**kwargs)
+        self.fields["file"].widget = ImageChoiceWidget()  # Filer misimplemented the field; we need to do this manually.
```
Comment (akx) [inline shoop/admin/modules/products/views/forms.py]: The comment is unnecessary (and not really correct here imho)

**Comment by akx** [comment]
Could you separate the additions into separate commits?

**Pull request merged**